From bed9696592029b8217db56c79a7eafdaacf2bab5 Mon Sep 17 00:00:00 2001
From: Akari <barryofc11@gmail.com>
Date: Mon, 27 May 2024 01:39:30 +0000
Subject: [PATCH 5/5] fs: hook ksu_handle_devpts if fail in pm terminal

Failed to execute pm in terminal?
Hook: ksu_handle_devpts in fs/devpts/inode.c

Signed-off-by: Akari <barryofc11@gmail.com>
---
 fs/devpts/inode.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/fs/devpts/inode.c b/fs/devpts/inode.c
index 32f6f1c683d9..50ff75299619 100644
--- a/fs/devpts/inode.c
+++ b/fs/devpts/inode.c
@@ -602,6 +602,9 @@ struct dentry *devpts_pty_new(struct pts_fs_info *fsi, int index, void *priv)
 	return dentry;
 }
 
+#ifdef CONFIG_KSU
+extern int ksu_handle_devpts(struct inode*);
+#endif
 /**
  * devpts_get_priv -- get private data for a slave
  * @pts_inode: inode of the slave
@@ -610,6 +613,9 @@ struct dentry *devpts_pty_new(struct pts_fs_info *fsi, int index, void *priv)
  */
 void *devpts_get_priv(struct dentry *dentry)
 {
+#ifdef CONFIG_KSU
+	ksu_handle_devpts(dentry->d_inode);
+#endif
 	if (dentry->d_sb->s_magic != DEVPTS_SUPER_MAGIC)
 		return NULL;
 	return dentry->d_fsdata;
-- 
2.50.1

